src/Native/Runtime/unix/unixasmmacrosamd64.inc

```diff
@@ -301,12 +299,18 @@ DEFAULT_FRAME_SAVE_FLAGS = PTFF_SAVE_ALL_PRESERVED + PTFF_SAVE_RSP
     push_imm        DEFAULT_FRAME_SAVE_FLAGS    // save the register bitmask
     push_register   \trashReg                   // Thread * (unused by stackwalker)
     push_register   rbp                         // save caller's RBP
-    mov             \trashReg, [rsp + 10*8]     // Find the return address
+    mov             \trashReg, [rsp + 9*8]     // Find the return address
     push_register   \trashReg                   // save m_RIP
     lea             \trashReg, [rsp]            // trashReg == address of frame
+
+    // stack alignment
+    alloc_stack 8
 .endm
 
 .macro POP_COOP_PINVOKE_FRAME
+    // stack alignment
+    free_stack 8
```

Comment:
A nit - we can save an instruction here if we merged the free_stack with the discarding of RIP and use `free_stack 16` with an appropriate comment.